src/lib/comms/sol-oic-client.c

```diff
@@ -57,6 +56,27 @@
 #define IOTIVITY_CON_REQ_OBS_MID 0x7d44
 #define IOTIVITY_NONCON_REQ_MID 0x7d40
 
+#define OIC_RESOURCE_CHECK_API(ptr, ...) \
```

Comment:
Since exported structures will always have an API version member and these checks will be performed, it might be a good idea to move this code to a general purpose set of macros.